src/main/java/Search/Searcher.java

```diff
@@ -0,0 +1,142 @@
+package Search;
+
+import Print.Printable;
+import Print.IntegerPair;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.TreeMap;
+
+import com.google.re2j.Matcher;
+import com.google.re2j.Pattern;
+
+public class Searcher {
+    private static final String DEFAULT_CHARSET = null; // Indicates platform default for FileUtils
+    private static final int PERIOD_MATCHES_NEWLINES_AND_SPACES = Pattern.DOTALL;
+    private static final String MATCH_INDICATOR = "||";
+    private static final char DELIMITER = ' ';
+
+    public Printable getPrintableSearchResult(Searchable searchable, String search) {
+        String transcript = getTranscript(searchable);
+        String timestamps = getTimestamps(searchable.timestampFile);
+        List<IntegerPair> matches = getMatches(transcript, search);
+        TreeMap<Integer, IntegerPair> transcriptTimestampIndices = mapTranscriptToTimestamps(transcript, timestamps);
+
+        return new Printable(matches, transcript, timestamps, searchable.filename, transcriptTimestampIndices);
+    }
+
+    private TreeMap<Integer, IntegerPair> mapTranscriptToTimestamps(String transcript, String timestamps) {
```

Comment:
Not a lot of "searching" going on here. Might be better for this to live in a different class. Maybe the printable itself, but might make the most sense to have it in whatever actually needs this map.

In its current state, you are creating some massive coupling unnecessarily:

Searcher has to pass map to Printable (printable constructor coupled with searcher).

Searcher creates map needed by Printer, even though searcher doesn't need the map and no Searcher specific information is used to create it. (Searcher directly coupled with Printer)

Reply:
Also, why return the integer pair index into the timestamps string, instead of just the timestamps substring that correspond to the integer? With the current setup, you have to store the entire timestamps string and all the integer pairs, and the map between the integers and integer pairs. You could change it so you just need a map between Integers (every index thats the start of a word in the transcript) and strings (every timestamp).

Reply:
Refactored both of these points, main issue now is finding a better way of printing N words before and after a match.